### DISCUSSION TITLE
batch save displaCy SVGs

### DISCUSSION THREAD

**Discussion by maddeldad** [explosion/spaCy#11875] [Help: Coding & Implementations]
I am looking at some tweet text data using spaCy. I would like to try some visualization experiments with displaCy image outputs. 

a simple case:

I would like batch of list of tweet texts into displaCy SVG diagrams, then potentially explore aligning/averaging those images as layers, just to see what kinds of interesting things emerge. perhaps some tic of language, some phrase or phrase structure, stands out visually under the right conditions.

for this purpose, it might be helpful first to textually align POS or other tags, DNA-style. I think I can wrangle that part, but with the images I am having trouble getting a single SVG to save. When I try to view the SVG output in a browsers, I see this message: "This XML file does not appear to have any style information associated with it. The document tree is shown below." 

I am using this method and can see the SVG in by notebook, but not in the save file:

https://stackoverflow.com/questions/56181929/save-spacy-render-file-as-svg-using-displacy

**Comment by polm** [comment]
If I do this:

```
import spacy
from spacy import displacy
nlp = spacy.load("en_core_web_sm")
doc = nlp("This is a sentence")
svg = displacy.render(doc)
with open("sample.svg", "w") as outfile:
    outfile.write(svg)
```

Then I can open that SVG file in Chrome without issue.

What code are you generating your SVG files with, and what browser are you using?

**Reply by serenalotreck** [reply]
Found this because I'm trying to output high resolution versions of my displacy visualizations for a paper. When I try this code, I get an error because the `svg` variable is `None`. I am in a Jupyter notebook, I imagine that's the issue, but do you know what I can do to get around it?

Thanks!

EDIT: I'm using `dep='ent'`

**Reply by polm** [reply]
See the note about [disabling Jupyter mode](https://spacy.io/usage/visualizers#jupyter) - you should be able to do that to get the SVG.

**Reply by serenalotreck** [reply]
I did try that and it does let me save, but I get a non-functional SVG that won't load in inkscape and doesn't appear as an image when I open it in Edge:
![WOS_000528876400009_displacy](https://github.com/explosion/spaCy/assets/41377532/342ff860-ad73-4d64-bbac-9d7eb95556dd)

**Reply by polm** [reply]
That output looks weird, not sure why that would happen. I would recommend you provide full, runnable code to reproduce the issue and open a new question.

**Reply by serenalotreck** [reply]
@polm I opened a new discussion for this with a minimal reproducible example! #13568
